CMakeLists.txt

```diff
@@ -78,6 +79,18 @@ IF(NNPACK_BACKEND STREQUAL "auto")
 ENDIF()
 
 IF(NNPACK_BACKEND STREQUAL "x86-64")
+  # ---[ CMAKE_SYSTEM_PROCESSOR will show x86_64 when we're compiling on 32 bit systems on 64 bit cpus
```

Comment:
These extra checks should be on line 73 where NNPACK automatically selects x86-64 backend, not here